src/juju/watchers.ts

```diff
@@ -49,23 +49,24 @@ export function generateModelWatcherBase(): ModelData {
   };
 }
 
-function _processDelta(
+function _processDelta<M extends ModelData, E extends keyof M>(
```

Comment:
This method has been updated to infer the types of the deltas from the new types provided by jujulib.